src/OrbitBase/include/OrbitBase/Align.h

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_ALIGN_H_
+#define ORBIT_BASE_ALIGN_H_
+
+#include "OrbitBase/Logging.h"
+
+namespace orbit_base {
+
+// alignment must be a power of 2
+template <uint64_t alignment>
+constexpr uint64_t AlignUp(uint64_t value) {
```

Comment:
nit. It would be nice to prevent implicit conversion here so that AlignUp<2>(-2) would not compile, but that seems more difficult than I thought. Do you know how that could be achieved?

Reply:
not really, @beckerhe may know if there is a way?

Reply:
The only solution I know of is making the parameter templated and using SFINAE. Roughly like so:
```
template <uint64_t alignment, typename T, typename = std::enable_if_t<std::is_same_v<T, uint64_t>>>
constexpr uint64_t AlignUp(T value) {
  static_assert((alignment & (alignment - 1)) == 0);
  return (value + (alignment - 1)) & ~(alignment - 1);
}
```

But that also means that `AlignUp<2>(2)` doesn't work anymore. You have to explicitly write `AlignUp<2>(2ul)`.

Reply:
yeah, I am not sure if I want to do that :)